src/frontend/src/App.js

```diff
@@ -8,30 +8,45 @@ import {LoginPage} from "./pages/LoginPage";
 import {ProfilePage} from "./pages/ProfilePage";
 import {CreatePage} from "./pages/CreatePage";
 import {SearchPage} from "./pages/SearchPage";
+<<<<<<< HEAD
+import Footer from "./components/footer";
+import { Switch } from "@mui/material";
+import { createContext, useState } from "react";
+export const ThemeContext = createContext(null);
+
+
+=======
 import Footer from "./components/Footer";
 import { ToastContainer, toast } from 'react-toastify';
 import 'react-toastify/dist/ReactToastify.css';
+>>>>>>> 0e4db4609b5b994e912d40f97c04ced782ed0ec8
 
 function App() {
-    if(localStorage.getItem("theme")==null)
-        localStorage.setItem("theme","light")
+    // if(localStorage.getItem("theme")==null)
+    //     localStorage.setItem("theme","light")
 
-    const changeTheme = (event)=>{
-        let theme = localStorage.getItem("theme")
-        if((theme==="dark")!==event.target.checked){
-            if(theme==="dark"){
-                theme = "light"
-            }
-            else{
-                theme = "dark"
-            }
-            localStorage.setItem("theme",theme)
-        }
-        console.log(localStorage.getItem("theme"))
-    }
+    // const changeTheme = (event)=>{
+    //     let theme = localStorage.getItem("theme")
+    //     if((theme==="dark")!==event.target.checked){
+    //         if(theme==="dark"){
+    //             theme = "light"
+    //         }
+    //         else{
+    //             theme = "dark"
+    //         }
+    //         localStorage.setItem("theme",theme)
+    //     }
+    //     console.log(localStorage.getItem("theme"))
+    const [theme, setTheme] = useState("light");
+    const toggleTheme = () => {
+    setTheme((curr) => (curr === "light" ? "dark" : "light"));
+        };
+    
     return (
-        <div className="App">
-            <Navbar changeTheme={changeTheme}/>
+        <ThemeContext.Provider value={{ theme, toggleTheme }}>
+        <div className="App" id={theme}>
+            <Navbar />
+            <Switch onChange={toggleTheme} checked={theme === "dark"} className="m-[10px] sm:m-0" />
```

Comment:
You need to keep this switch in the navbar not here, you can pass the toggle theme as a prop to the navbar